### ISSUE TITLE
Sun's Corona requires a Fire Beetle Eye to cast

### ISSUE THREAD

**Issue by mackal** [EQEmu/Server#623]
My druid can cast this spell line just fine without one in her inventory, but live doesn't let me without having one!

**Comment by daerath** [comment]
Can you re-confirm?  Alla's lists the Fire Beetle Eye as a Focus and not as a Reagent.

**Comment by mackal** [comment]
Alla is silly is my guess. Looking at PEQ editor Sun's Corona has Fire Beetle Eye in "No Expend Reagent 1" this means it requires a Fire Beetle Eye present to cast (which isn't consumed)

You need the item in your inventory, not bank :P

**Comment by daerath** [comment]
Gotcha.  Does it get used up on live?

**Comment by mackal** [comment]
No, you just need it

**Comment by daerath** [comment]
I'm confused then.  When the ID for a fire beetle eye is added to the components1 value for that spell, attempts to cast it fail if you don't have it in your inventory.  If it is also listed as a noexpendreagent1 it doesn't get consumed.

Is the real issue that the emu should also be checking the noexpendreagents values (and it isn't)?

**Comment by mackal** [comment]
Because that's not how it's suppose to work. Without changing the spell data, it should work as I said.

I haven't actually look far enough into fixing, why I opened the issue.

**Comment by daerath** [comment]
Ok.  I'll work under the assumptuon that the spell data is correct.  I'll look at the code tomorrow.

**Comment by daerath** [comment]
I see the issue in the code and also confirmed this by testing Banshee Aura.  It also requires a component (Pearl, not consumed), but the Emu doesn't care if you have one or not.  Data looks the same (-1s for components, but the ID for Pearl is listed as a non-consumable).

**Comment by mackal** [comment]
Make sure you don't break bard songs! (main reason I haven't bothered)

**Comment by mackal** [comment]
Okay, so it looks like from the details I can gleam from the client, bards have special behavior here. The NOEXPENDREAGENT for bards is kind of just convention and doesn't act the same for other clients :P (they're actually checking the spell's skill to get the item type needed)

**Comment by daerath** [comment]
Yup, I kept the bard bits mostly the same.  I made one minor change (removed it from a loop that made no sense because it just did the exact same check each iteration).  Going to do some playtesting over the next couple of days to make sure it works.

**Comment by daerath** [comment]
Withdrew pull request.
